From 8c9437394e967ce02345be682734e8f9707feee1 Mon Sep 17 00:00:00 2001
From: Kylie McClain <somasis@exherbo.org>
Date: Wed, 11 May 2016 08:05:43 -0400
Subject: [PATCH 3/5] Add fortify-headers support for non-glibcs

On non-glibc, fortify functionality needs to be provided by
sys-libs/fortify-headers; this makes it included by default so that
it can be activated the usual way, by just passing
-D_FORTIFY_SOURCE=[0|1|2].
---
 gcc/config/linux.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gcc/config/linux.h b/gcc/config/linux.h
index 95654bcdb5a..d88df8f154a 100644
--- a/gcc/config/linux.h
+++ b/gcc/config/linux.h
@@ -167,6 +167,8 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 
 #ifdef NATIVE_SYSTEM_HEADER_DIR
 #define INCLUDE_DEFAULTS_MUSL_NATIVE			\
+    { NATIVE_SYSTEM_HEADER_DIR "/fortify", 0, 0, 0, 1, 2 },	\
+    { NATIVE_SYSTEM_HEADER_DIR "/fortify", 0, 0, 0, 1, 0 },	\
     { NATIVE_SYSTEM_HEADER_DIR, 0, 0, 0, 1, 2 },	\
     { NATIVE_SYSTEM_HEADER_DIR, 0, 0, 0, 1, 0 },
 #else
-- 
2.26.2

